lib/msf/base/sessions/command_shell.rb

```diff
@@ -654,6 +654,7 @@ def shell_command(cmd, timeout=5)
   def shell_read(length=-1, timeout=1)
     begin
       rv = rstream.get_once(length, timeout)
+      rlog(rv, self.log_source) if rv && self.log_source
```

Comment:
It looks like the `rlog` pattern was introduced by skape way-back-when https://github.com/rapid7/metasploit-framework/commit/db40a555479577a44e50383d43b0216175db5a88 which writes to `~/.msf4/sessions`

But then the newer pattern below that uses the framework event bus `framework.events.on_session_output(self, rv)` was added for persisting logs to the database.